airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/api.py

```diff
@@ -87,6 +89,12 @@ def _parse_call_rate_header(headers):
 
         return usage, pause_interval
 
+    @staticmethod
+    def sleep_time_handler(usage, pause_interval: pendulum.duration):
+
```

Comment:
I think the definition of the custom pause interval according to `usage` should be determined in this function and not in `handle_call_rate_limit`.
I'd rename it to `compute_pause_interval` and make it return the total seconds to sleep. Then I'd call sleep with this value from `handle_call_rate_limit`. I think it would greatly improve the readability and maintainability of this logic.

Reply:
@alafanechere Thanks, I did some changes, tell me what you think.